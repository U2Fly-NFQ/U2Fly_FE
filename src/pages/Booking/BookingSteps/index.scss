.booking-steps__flight {
  &__container {
    display: flex;
    // gap: 0.5rem;
    flex-direction: column;
  }
  &__content {
    display: flex;
    gap: 0.5rem;
  }
}
