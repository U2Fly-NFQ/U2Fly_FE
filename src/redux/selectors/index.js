export * from './auth'
export * from './bookingFlightSelector'
export * from './ticket'
