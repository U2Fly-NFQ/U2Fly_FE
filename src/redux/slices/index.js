export * from './searchSlice'
export * from './filterSlice'
export * from './flightSlice'
