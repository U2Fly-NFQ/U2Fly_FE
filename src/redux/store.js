import { configureStore } from '@reduxjs/toolkit'
import bookingFlightsSlice from './slices/bookingFlightsSlice'

// import loginSlice from './slices/loginSlice'
import flightSlice from './slices/flightSlice'
import airportSlice from './slices/airportSlice'

const store = configureStore({
  reducer: {
    // login: loginSlice.reducer,
    bookingFlight: bookingFlightsSlice.reducer,
    flights: flightSlice.reducer,
    airports: airportSlice.reducer,
  },
})

export default store
