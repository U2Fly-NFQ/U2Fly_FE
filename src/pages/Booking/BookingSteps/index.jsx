import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Steps } from 'antd'
import { getCurrentMethodInBookingFlight } from '../../../redux/selectors'
import './index.scss'
import { changeCurrentMethod } from '../../../redux/slices/bookingFlightsSlice'

export default function BookingSteps({ contentTop, contentBottom, ticketId }) {
  const dispatch = useDispatch()
  const getCurrentMethod = useSelector(getCurrentMethodInBookingFlight)

  const { Step } = Steps
  const onChange = (value) => {
    dispatch(changeCurrentMethod(value))
  }

  return (
    <Steps
      type="navigation"
      className="booking-steps"
      current={ticketId ? 2 : getCurrentMethod}
      // current={0}
    >
      <Step
        className="booking-steps__flight"
        title={
          <div className="booking-steps__flight__container">
            <div className="booking-steps__flight__logo">
              <i class="fa-regular fa-user"></i>
            </div>
            <div>
              <div className="booking-steps__flight__content">
                <h4>Thông tin khách Hàng</h4>
              </div>
            </div>
          </div>
        }
      />
      <Step
        title={
          <div className="booking-steps__flight__container">
            <div className="booking-steps__flight__logo">
              <i class="fa-solid fa-credit-card"></i>
            </div>
            <div>
              <div className="booking-steps__flight__content">
                <h4>Phương Thức Thanh Toán</h4>
              </div>
            </div>
          </div>
        }
      />
      <Step
        status="wait"
        title={
          <div className="booking-steps__flight__container">
            <div className="booking-steps__flight__logo">
              <i class="fa-solid fa-circle-check"></i>
            </div>
            <div>
              <div className="booking-steps__flight__content">
                <h4>Hoàn tất thanh toán</h4>
              </div>
            </div>
          </div>
        }
      />
    </Steps>
  )
}
