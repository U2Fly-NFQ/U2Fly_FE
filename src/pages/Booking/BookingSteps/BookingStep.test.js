import { I18nextProvider } from 'react-i18next'
import i18n from '../../../translations'
import renderer from 'react-test-renderer'
import React, { Suspense } from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import { store } from '../../../redux/store'
import BookingSteps from './'

describe('Booking Passenger bar test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <Router>
              <Suspense>
                <BookingSteps />
              </Suspense>
            </Router>
          </I18nextProvider>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
