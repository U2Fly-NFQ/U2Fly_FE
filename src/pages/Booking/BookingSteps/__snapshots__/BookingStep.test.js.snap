// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Passenger bar test renders correctly 1`] = `
<div
  className="ant-steps ant-steps-horizontal booking-steps ant-steps-label-horizontal ant-steps-navigation"
  style={Object {}}
>
  <div
    className="ant-steps-item ant-steps-item-process booking-steps__flight ant-steps-item-active"
    style={Object {}}
  >
    <div
      className="ant-steps-item-container"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="ant-steps-item-tail"
      />
      <div
        className="ant-steps-item-icon"
      >
        <span
          className="ant-steps-icon"
        >
          1
        </span>
      </div>
      <div
        className="ant-steps-item-content"
      >
        <div
          className="ant-steps-item-title"
        >
          <div
            className="booking-steps__flight__container"
          >
            <div
              className="booking-steps__flight__logo"
            >
              <i
                className="fa-regular fa-user"
              />
            </div>
            <div>
              <div
                className="booking-steps__flight__content"
              >
                <h4>
                  Passenger information
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-steps-item ant-steps-item-wait"
    style={Object {}}
  >
    <div
      className="ant-steps-item-container"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="ant-steps-item-tail"
      />
      <div
        className="ant-steps-item-icon"
      >
        <span
          className="ant-steps-icon"
        >
          2
        </span>
      </div>
      <div
        className="ant-steps-item-content"
      >
        <div
          className="ant-steps-item-title"
        >
          <div
            className="booking-steps__flight__container"
          >
            <div
              className="booking-steps__flight__logo"
            >
              <i
                className="fa-solid fa-credit-card"
              />
            </div>
            <div>
              <div
                className="booking-steps__flight__content"
              >
                <h4>
                  Payment methods
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-steps-item ant-steps-item-wait"
    style={Object {}}
  >
    <div
      className="ant-steps-item-container"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="ant-steps-item-tail"
      />
      <div
        className="ant-steps-item-icon"
      >
        <span
          className="ant-steps-icon"
        >
          3
        </span>
      </div>
      <div
        className="ant-steps-item-content"
      >
        <div
          className="ant-steps-item-title"
        >
          <div
            className="booking-steps__flight__container"
          >
            <div
              className="booking-steps__flight__logo"
            >
              <i
                className="fa-solid fa-circle-check"
              />
            </div>
            <div>
              <div
                className="booking-steps__flight__content"
              >
                <h4>
                  payment successful
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
